tests/tests/eth_provider.rs

```diff
@@ -805,6 +808,260 @@ async fn test_send_raw_transaction_wrong_signature(#[future] katana: Katana, _se
+    };
+
+    // Perform the call to invoke the incForLoop function without state override
+    // Attest that the transaction succeeds
+    let _ = eth_provider
+        .call(request.clone(), None, None, None)
+        .await
+        .expect("Failed to call incForLoopCall in Kakarot constract");
+
+    // Define another Solidity contract with a different interface and bytecode
+    sol! {
+        #[sol(rpc, bytecode = "60803461006357601f61014838819003918201601f19168301916001600160401b038311848410176100675780849260209460405283398101031261006357518015158091036100635760ff80195f54169116175f5560405160cc908161007c8239f35b5f80fd5b634e487b7160e01b5f52604160045260245ffdfe60808060405260043610156011575f80fd5b5f3560e01c9081638bf1799f14607a575063b09a261614602f575f80fd5b346076576040366003190112607657602435801515810360765715606f57604060015b81516004356001600160a01b0316815260ff919091166020820152f35b60405f6052565b5f80fd5b346076575f36600319011260765760209060ff5f541615158152f3fea264697066735822122043709781c9bdc30c530978abf5db25a4b4ccfebf989baafd2ba404519a7f7e8264736f6c63430008180033")]
+        contract MyContract {
+            bool public myState;
+
+            constructor(bool myState_) {
+                myState = myState_;
+            }
+
+            function doStuff(uint a, bool b) external pure returns(address c, bytes32 d) {
+                return (address(uint160(a)), bytes32(uint256(b ? 1 : 0)));
+            }
+        }
+    }
+
+    // Extract the bytecode for the MyContract contract
+    let bytecode = &MyContract::BYTECODE[..];
+
+    // State override with the MyContract bytecode
+    // Change the contract bytecode
+    let mut state_override: HashMap<Address, AccountOverride> = HashMap::new();
+    state_override.insert(
+        contract_address,
+        AccountOverride { code: Some(bytecode.into()), balance: Some(U256::MAX), ..Default::default() },
+    );
+
+    // Attempt to call the incForLoop function which doesn't exist in the MyContract contract
+    //
+    // Should fail as the contract bytecode is different now with state override
+    match eth_provider.call(request, None, Some(state_override), None).await {
+        // If call succeeds, panic as an error was expected
+        Ok(_) => panic!("Expected an error but got a success response"),
+        // If call fails, check if the error is due to a transaction revert
+        Err(e) => match e {
+            EthApiError::RethEthApi(RethEthApiError::InvalidTransaction(RpcInvalidTransactionError::Revert(err))) => {
+                assert_eq!(err.to_string(), RevertError::new(Default::default()).to_string());
+            }
+            // If a different error occurs, panic
+            _ => panic!("Expected an invalid transaction revert error but got a different error: {e:?}"),
+        },
+    }
+}
+
+#[rstest]
+#[awt]
+#[tokio::test(flavor = "multi_thread")]
+async fn test_call_with_state_override_bytecode1(#[future] plain_opcodes: (Katana, KakarotEvmContract), _setup: ()) {
```

Comment:
I'm not sure we need this extract test, I don't think it brings much more over the previous test

Reply:
Mhhh not sure, maybe better to delete the first one but in my opinion, knowing that it's quite complex and that it doesn't add too much cost I would keep both.
- The firs tests that if I deploy a smart contract and I call a function inside it works (normal). But if I override the smart contract with another bytecode then this same call no longer works (the override has taken place).
- This second test that you point out is a little different because it verifies that if I deploy a smart contract and I change the bytecode directly to mimic another smart contract then this time I have the right to call a function of this new smart contract and it works (the override worked well and the bytecode read corresponds to what I want for my new smart contract).

So I would say that the second test is more powerful and if I had to keep only one I would keep this one. As you prefer, if you prefer to remove the first one for CI costs then let's do it :)

Reply:
I don't really see the difference in the two tests, for me it's just testing that override works, once with a call before to the same contract and the second test without this extra test. I would remove at least one

Reply:
Just removed the first one, as the second more powerful (testing the call to a new function of the override).